web/src/main/java/org/hillview/InitialObjectTarget.java

```diff
@@ -30,18 +35,47 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 import java.util.logging.Level;
 import java.util.stream.Stream;
 
 public class InitialObjectTarget extends RpcTarget {
+    private static final int HILLVIEW_DEFAULT_PORT = 3569;
+    private static final String LOCALHOST = "127.0.0.1";
+
     @Nullable
-    private final IDataSet<Empty> emptyDataset;
+    private IDataSet<Empty> emptyDataset = null;
 
     InitialObjectTarget() {
         Empty empty = new Empty();
-        this.emptyDataset = new LocalDataSet<Empty>(empty);
+        final List<HostAndPort> hostAndPorts = Collections.singletonList(
+            HostAndPort.fromParts(LOCALHOST, HILLVIEW_DEFAULT_PORT)
+        );
+        final ClusterDescription desc = new ClusterDescription(hostAndPorts);
+        this.initialize(desc);
+    }
+
+    private void initialize(final ClusterDescription description) {
+        final int numServers = description.getServerList().size();
+        assert numServers > 0;
```

Comment:
This should not be an assert; asserts are only for internal logic bugs, but this is input validation.
The cluster description could come from a user-supplied file.